const statusCode = require("../../core/status/statusCode");
const ratingReviewServices = require("../../core/services/ratingReviewServices/ratingReviewServices");

const createErrorMessage = (message, data) => {
  return {
    status: statusCode,
    data: data,
    message: message,
    error: true,
  };
};

module.exports = {
  async createRatingReviewController(req, res) {
    try {
      let response = await ratingReviewServices.createRatingReviewService(req.body);
      return res.status(response.status).send(response);
    } catch (error) {
      console.error(error);
      const newError = createErrorMessage();
      newError.data = error;
      newError.message = "Create ratingReview Controller Error";
      newError.status = statusCode.internalServerError;
      newError.error = true;
      return res.status(newError.status).json(newError);
    }
  },

  async getAllRatingReviewController(req, res) {
    try {
      let response = await ratingReviewServices.getAllRatingReviewService();
      return res.status(response.status).send(response);
    } catch (error) {
      console.error(error);
      const newError = createErrorMessage();
      newError.data = error;
      newError.message = "Create ratingReview Controller Error";
      newError.status = statusCode.internalServerError;
      newError.error = true;
      return res.status(newError.status).json(newError);
    }
  },

  async getSingleRatingReviewController(req, res) {
    try {
      let response = await ratingReviewServices.getSingleRatingReviewService(req.params);
      return res.status(response.status).send(response);
    } catch (error) {
      console.error(error);
      const newError = createErrorMessage();
      newError.data = error;
      newError.message = "Create ratingReview Controller Error";
      newError.status = statusCode.internalServerError;
      newError.error = true;
      return res.status(newError.status).json(newError);
    }
  },

  async getSingleUserRatingReviewController(req, res) {
    try {
      let response = await ratingReviewServices.getSingleUserRatingReviewService(req.params);
      return res.status(response.status).send(response);
    } catch (error) {
      console.error(error);
      const newError = createErrorMessage();
      newError.data = error;
      newError.message = "Create ratingReview Controller Error";
      newError.status = statusCode.internalServerError;
      newError.error = true;
      return res.status(newError.status).json(newError);
    }
  },
  async updateSingleRatingReviewController() {
    try {
      let response = await ratingReviewServices.updateSingleRatingReviewService(req.params, req.body);
      return res.status(response.status).send(response);
    } catch (error) {
      console.error(error);
      const newError = createErrorMessage();
      newError.data = error;
      newError.message = "Create ratingReview Controller Error";
      newError.status = statusCode.internalServerError;
      newError.error = true;
      return res.status(newError.status).json(newError);
    }
  },
  async removeAllRatingReviewController() {
    try {
      let response = await ratingReviewServices.removeAllRatingReviewService();
      return res.status(response.status).send(response);
    } catch (error) {
      console.error(error);
      const newError = createErrorMessage();
      newError.data = error;
      newError.message = "Create ratingReview Controller Error";
      newError.status = statusCode.internalServerError;
      newError.error = true;
      return res.status(newError.status).json(newError);
    }
  },
  async removeSingleRatingReviewController(req, res) {
    try {
      let response = await ratingReviewServices.removeSingleRatingReviewService(req.params);
      return res.status(response.status).send(response);
    } catch (error) {
      console.error(error);
      const newError = createErrorMessage();
      newError.data = error;
      newError.message = "Create ratingReview Controller Error";
      newError.status = statusCode.internalServerError;
      newError.error = true;
      return res.status(newError.status).json(newError);
    }
  },
};
